modules/services/xcape.nix

```diff
@@ -0,0 +1,77 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.xcape;
+
+in
+
+{
+  meta.maintainers = [ maintainers.nickhu ];
+
+  options = {
+    services.xcape = {
+      enable = mkEnableOption "xcape";
+
+      package = mkOption {
+        default = pkgs.xcape;
+        defaultText = "pkgs.xcape";
+        type = types.package;
+        example = literalExample "pkgs.xcape";
+        description = "The package to use for the xcape binary.";
+      };
+
+      timeout = mkOption {
+        type = types.nullOr types.int;
+        default = null;
+        defaultText = "<timeout ms>";
+        example = 500;
+        description = "If you hold a key longer than this timeout, xcape will not generate a key event. Default is 500 ms.";
+      };
+
+      mapExpression = mkOption {
+        type = types.attrsOf types.str;
+        default = {};
+        example = { Shift_L = "Escape"; Control_L = "Control_L|O"; };
+        description = ''
+          The expression has the grammar 'ModKey=Key[|OtherKey][;NextExpression]'
+
+          The list of key names is found in the header file X11/keysymdef.h
+          (remove the XK_ prefix). Note that due to limitations of X11 shifted
+          keys must be specified as a shift key followed by the key to be
+          pressed rather than the actual name of the character. For example to
+          generate "{" the expression 'ModKey=Shift_L|bracketleft' could be used
+          (assuming that you have a key with "{" above "[").
+
+          You can also specify keys in decimal (prefix #), octal (#0), or
+          hexadecimal (#0x). They will be interpreted as keycodes unless no
+          corresponding key name is found.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.user.services.xcape = {
+        Unit = {
+          Description = "xcape allows you to use a modifier key as another key when pressed and released on its own";
+          After = [ "graphical-session-pre.target" ];
+          PartOf = [ "graphical-session.target" ];
+        };
+
+        Service = {
+          Type = "forking";
```

Comment:
Maybe it's possible to remove this if the `-f` option is used? The xcape readme says "Foreground mode. Does not fork into the background."

Reply:
Sure - is there any reason non-forking is preferred in systemd units?

Reply:
I think non-forking is preferred because it is much simpler. And systemd assumes non-forking by default..

Reply:
Ok, I have made it non-forking in the latest commit :)